cloud/pkg/controller/controller/downstream.go

```diff
@@ -165,6 +172,35 @@ func (dc *DownstreamController) syncSecret(stop chan struct{}) {
 	}
 }
 
+func (dc *DownstreamController) syncEdgeNodes(stop chan struct{}) {
+	running := true
+	for running {
+		select {
+		case e := <-dc.nodeManager.Events():
+			node, ok := e.Object.(*v1.Node)
+			if !ok {
+				log.LOGGER.Warnf("object type: %T unsupported", node)
+				continue
+			}
+			switch e.Type {
+			case watch.Added:
+				dc.lc.UpdateEdgeNode(node.ObjectMeta.Name)
+			case watch.Modified:
+				continue
```

Comment:
We need to add modified case, if the role a node is changed to EdgeNode,we need to process it and vice versa

Reply:
Whenever change in role label added and deleted case is happening so no need to see for modified requests.

Reply:
NIce Catch!

Reply:
> Whenever change in role label added and deleted case is happening so no need to see for modified requests.

Sorry I sitll don't understand why changing the node labels will produce add or delete events. We had better to write some UTs for this function.

Reply:
@m1093782566 , The informer is started with a label selector with edge as role. So when the role is added to a node, it will be added case, If the role is changed/removed, it will be a deleted case. The other updates in the spec/status of the node doesn't effect the role of the node hence the update case can be left empty....